### ISSUE TITLE
Display of readme.txt in Package Manager seriously crippled.

### ISSUE THREAD

**Issue by BobRay** [modxcms/revolution#16206]
### Summary
When installing a package in Package Manager, the rendering of Markdown content is seriously strange when viewing the Readme tab.
 
 - There's no apparent way to separate paragraphs. All newlines are reduced to a single newline so there is no space between paragraps (and as far as I can tell, there's no way to separate them).
 - Backticks are either ignored and removed, or rendered literally.
 - Text surrounded by single backticks is rendered as regular text, rather than highlighted monospace.
 - As far as I can tell, all HTML tags outside of code sections are rendered literally, as are entities. 
 - Opening HTML beginning tag tokens are rendered as literal entities (&lt ;) inside of backticks.
 - Code sections between three backticks are rendered in the regular text font with no highlighting.

### Step to reproduce
It's difficult to illustrate the problem here, but try this code in a readme.txt for a package (without the triple backticks at each end), then look at the Readme tab when installing it. Add a code section between sets of three backticks somewhere below.

```
<p>line1</p>


<p>Line2</p>



Line 3.

Put this tag (`<br/>`) in the `<head>` section of your template.

&nbsp;&nbsp;
<p>&nbsp;</p>
<br />
<br>
Some text.



xxx
```

### Observed behavior
Output is nothing like what you would expect.

### Expected behavior
Should be rendered like normal Markdown with HTML tags outside of code sections interpreted as HTML tags.

### Environment
MODX 3.0.1 makes more of a mess of things.
MODX 2.8.4 also renders backticks literally, but does not convert opening HTML tag tokens as entities. It preserves space between paragraphs, but it also renders text between single backticks the same as regular text (and displays the backticks). It also does not display code between three backticks as monospace and without any highlighting.

**Comment by BobRay** [comment]
The code above, and the code I've created for packages is rendered perfectly at https://parsedown.org/demo.

Also, MODX 2.8.4, which uses Parsedown 1.0.0 doesn't parse links correctly.

**Comment by Mark-H** [comment]
If this is about how it's shown, screenshots would be useful. Wouldn't be the first time the mess that is ExtJS has styles that break plain markup!

If it's actually parsing the content wrong, then that's certainly peculiar.  

Is this from a specific package people can try to install for testing? I doubt anyone is going to build an extra just to test this out.

**Comment by BobRay** [comment]
Any extra will do, at least on my local machine. Click on Install or Re-Install, then click on the readme tab. The Login package is probably the most obvious example. [Here's a link to a screenshot](https://bobsguides.com/assets/downloads/Login_Screenshot.jpg).

I think all three installation tabs are being rendered as plain text, no matter what the content is or what the file extension is.  HTML tags, like the backticks in the screenshot, are rendered literally and have no effect on the output. The effect is the same in Chrome and Firefox.

**Comment by Ruslan-Aleev** [comment]
There is another problem: the **first** text field (apparently set to :focus) in the package section is always scrolled down, which is inconvenient.
![first_textarea](https://user-images.githubusercontent.com/12523676/176379601-08e0c93f-ca60-44ee-a7e7-46f3df4ece0b.png)

**Comment by Mark-H** [comment]
Hm, it looks to me like it should always be parsed as markdown: https://github.com/modxcms/revolution/blob/3.x/core/src/Revolution/Processors/Workspace/Packages/GetAttribute.php#L81

Maybe the htmlentities is breaking things? Of course we do want to avoid executing arbitrary code from a changelog or readme... but that's what safemode is probably for.

**Comment by BobRay** [comment]
It's definitely the `htmlentities()` call (though I can't see why). Removing it returns things to normal.  I was unable to find the equivalent code in MODX 2.8.4, but it has the same problem.

**Comment by smg6511** [comment]
Yes, `htmlspecialchars` is the cause of the main problem. In 3.x **Parsedown** is used to convert **Markdown** to html (I'm guessing this was not the case in 2.x), so there is no need to use html tags in the _change log/readme/license_ texts (at least in the package manifest). More importantly we're using safe mode in Parsedown, which does what `htmlspecialchars` does and more (such as keeping js/XSS from being introduced in a Markdown link). My guess is that the idea was to be able to easily port over your `.md` content as-is and have it display nicely in the manager. 

That said, it seems like that idea has not been carried through to other areas where this content is used (_i.e._, within “View Details” in the Package Browser and the Packages list). 

Two current PRs address the display issues (#16829 and #16952); take a look at those and please review ;-)
